BTrees/_base.py

```diff
@@ -40,6 +40,44 @@ def __init__(self, items=None):
         if items:
             self.update(items)
 
+    try:
+        # Detect the presence of the C extensions.
+        # If they're NOT around, we don't need to do any of the
+        # special pickle support to make Python versions look like
+        # C---we just rename the classes. By not defining these methods,
+        # we can (theoretically) avoid a bit of a slowdown.
+        # If the C extensions are around, we do need these methods, but
+        # these classes are unlikely to be used in production anyway.
+        __import__('BTrees._OOBTree')
+    except ImportError:
+        pass
+    else:
+        def __reduce__(self):
+            # Swap out the type constructor for the C version, if present.
+            func, typ_gna, state = Persistent.__reduce__(self)
+            # We ignore the returned type altogether in favor of
+            # our calculated class (which allows subclasses but replaces our exact
+            # type with the C equivalent)
+            typ = self.__class__
+            gna = typ_gna[1:]
+            return (func, (typ,) + gna, state)
+
+        @property
+        def __class__(self):
+            type_self = type(self)
+            return type_self._BTree_reduce_as if type_self._BTree_reduce_up_bound is type_self else type_self
+
+        @property
+        def _BTree_reduce_as(self):
+            # Return the pickle replacement class for this object.
+            # If the C extensions are available, this will be the
+            # C type (setup by _fix_pickle), otherwise it will be the real
+            # type of this object.
+            # This implementation is replaced by _fix_pickle and exists for
+            # testing purposes.
+            return type(self)
```

Comment:
There is a mechanism for sharing coverage data between tox environments.  E.g., Pyramid uses both python2 and python3 for coverage:

  https://github.com/Pylons/pyramid/blob/master/tox.ini#L84